domain/domain.go

```diff
@@ -256,20 +270,63 @@ func (c *ddlCallback) OnChanged(err error) error {
 	}
 	log.Warnf("[ddl] on DDL change")
 
-	c.do.mustReload()
-	return nil
+	return c.do.MustReload()
+}
+
+type schemaValidityInfo struct {
+	validity         bool
+	lastInvalidTS    uint64
+	mux              sync.RWMutex
+	txnTS            uint64 // It's used for checking schema validity.
```

Comment:
This field can be removed, we can use `lastInvalidTs` to make sure old transaction must fail after failed to load information schema.